src/confluent_kafka/schema_registry/avro.py

```diff
@@ -200,9 +224,9 @@ def __init__(self, schema_registry_client, schema_str,
             raise ValueError("Unrecognized properties: {}"
                              .format(", ".join(conf_copy.keys())))
 
-        schema = _schema_loads(schema_str)
         schema_dict = loads(schema.schema_str)
-        parsed_schema = parse_schema(schema_dict)
+        self._named_schemas = _resolve_named_schema(schema, schema_registry_client)
+        parsed_schema = parse_schema(schema_dict, named_schemas=self._named_schemas)
```

Comment:
Aren't these two lines doing the same thing?

Reply:
`_resolve_named_schema` is parsing all the referenced schemas, but not the top level one. Line number 229 parses the top level one using the `named_schemas`.

Reply:
Also, parse_schema adds the parsed schema to the named_schemas dict